hadoop-hdds/container-service/src/test/java/org/apache/hadoop/ozone/container/common/TestBlockDeletingService.java

```diff
@@ -158,65 +190,167 @@ private void createToDeleteBlocks(ContainerSet containerSet,
           BlockID blockID =
               ContainerTestHelper.getTestBlockID(containerID);
-          String deleteStateName = OzoneConsts.DELETING_KEY_PREFIX +
-              blockID.getLocalID();
           BlockData kd = new BlockData(blockID);
           List<ContainerProtos.ChunkInfo> chunks = Lists.newArrayList();
-          for (int k = 0; k < numOfChunksPerBlock; k++) {
-            final String chunkName = String.format("block.%d.chunk.%d", j, k);
-            final long offset = k * chunkLength;
-            ContainerProtos.ChunkInfo info =
-                ContainerProtos.ChunkInfo.newBuilder()
-                    .setChunkName(chunkName)
-                    .setLen(chunkLength)
-                    .setOffset(offset)
-                    .setChecksumData(Checksum.getNoChecksumDataProto())
-                    .build();
-            chunks.add(info);
-            ChunkInfo chunkInfo = new ChunkInfo(chunkName, offset, chunkLength);
-            ChunkBuffer chunkData = buffer.duplicate(0, (int) chunkLength);
-            chunkManager.writeChunk(container, blockID, chunkInfo, chunkData,
-                WRITE_STAGE);
-            chunkManager.writeChunk(container, blockID, chunkInfo, chunkData,
-                COMMIT_STAGE);
-          }
+          putChunksInBlock(numOfChunksPerBlock, i, chunks, buffer, chunkManager,
+              container, blockID);
           kd.setChunks(chunks);
-          metadata.getStore().getBlockDataTable().put(
-                  deleteStateName, kd);
+          try(ReferenceCountedDB metadata = BlockUtils.getDB(data, conf)) {
+            String bID = blockID.getLocalID()+"";
+            metadata.getStore().getBlockDataTable().put(bID, kd);
+          }
           container.getContainerData().incrPendingDeletionBlocks(1);
+
+          // In below if statements we are checking if a single container
+          // consists of more blocks than 'blockLimitPerTask' then we create
+          // (totalBlocksInContainer / blockLimitPerTask) transactions which
+          // consists of blocks equal to blockLimitPerTask and last transaction
+          // consists of blocks equal to
+          // (totalBlocksInContainer % blockLimitPerTask).
+          if (blockCount < blockLimitPerTask) {
+            if (containerBlocks.containsKey(containerID)) {
+              containerBlocks.get(containerID).add(blockID.getLocalID());
+            } else {
+              List<Long> item = new ArrayList<>();
+              item.add(blockID.getLocalID());
+              containerBlocks.put(containerID, item);
+            }
+            blockCount++;
+          }
+          boolean flag = false;
+          if (blockCount == blockLimitPerTask) {
+            createTxn(data, containerBlocks, txnID);
+            containerBlocks.clear();
+            blockCount = 0;
+            flag = true;
+          }
+          if (i == (numOfBlocksPerContainer - 1) && !flag) {
+            createTxn(data, containerBlocks, txnID);
+          }
```

Comment:
```suggestion
          containerBlocks.add(blockID.getLocalID());
          blockCount++;
          if (blockCount == blockLimitPerTask || i == (numOfBlocksPerContainer - 1)) {
            createTxn(data, containerBlocks, txnID);
            containerBlocks.clear();
            blockCount = 0;
            flag = true;
          }
```
I think we can simplify the code here. We can use a List for storing containerBlocks.

Reply:
Thanks for this suggestion, made the necessary changes.